include/configoptions.h

```diff
@@ -0,0 +1,361 @@
+        */
+        void override_integer_option(const std::string& key, int64_t value);
+
+        /*!
+        *   \brief Override the value of a string configuration option
+        *          in the selected source
+        *   \details Overrides are specific to an instance of the
+        *            ConfigOptions class. An instance that references
+        *            the same source will not be affected by an override to
+        *            a different ConfigOptions instance
+        *   \param key The name of the configuration option to override
+        *   \param value The value to store for the configuration option
+        */
+        void override_string_option(const std::string& key, const std::string& value);
+
+        /*!
+        *   \brief Override the value of a boolean configuration option
+        *          in the selected source
+        *   \details Overrides are specific to an instance of the
+        *            ConfigOptions class. An instance that references
+        *            the same source will not be affected by an override to
+        *            a different ConfigOptions instance
+        *   \param key The name of the configuration option to override
+        *   \param value The value to store for the configuration option
+        */
+        void override_boolean_option(const std::string& key, bool value);
+
+    private:
+
+        /*!
+        *   \brief Process option data from a fixed source
+        *   \throw SmartRedis::Exception: sources other than environment
+        *          variables are not implemented yet
+        */
+        void _populate_options();
+
+        /*!
+        *   \brief Apply a prefix to a key if the source is environment
+        *          variables and the prefix is nonempty
+        *   \param key The key to prefix
+        */
+        std::string _prefixed(const std::string& key);
+
+        /*!
+        *  \brief Boolean option map
+        */
+        std::unordered_map<std::string, bool> _bool_options;
+
+        /*!
+        *  \brief Integer option map
+        */
+        std::unordered_map<std::string, int64_t> _int_options;
+
+        /*!
+        *  \brief String option map
+        */
+        std::unordered_map<std::string, std::string> _string_options;
+
+        /*!
+        *  \brief Configuration source
+        */
+        cfgSrc _source;
+
+        /*!
+        *  \brief Configuration string. Meaning is specific to the source
+        */
+       std::string _string;
+
+        /*!
+        *  \brief Lazy evaluation (do we read in all options at once or only
+        *         on demand)
+        */
+        bool _lazy;
+
+        /*!
+        *  \brief Logging context
+        */
+        std::string _log_context;
+
+        /*!
+        *  \brief Default configuration source
+        */
+        static cfgSrc _default_source;
+
+        /*!
+        *  \brief Default configuration string -- meaning is specific to the
```

Comment:
The "For from..." construct is a little difficult to parse, maybe just refer to the enums?

Reply:
Fixed, yeah, that wasn't nearly as clear on re-read as when I wrote it